backend/handlers/math/outcomes/dbpm/marketshares_test.go

```diff
@@ -1,551 +1,709 @@
 package dbpm
 
 import (
-	"math"
 	"reflect"
 	"socialpredict/handlers/math/probabilities/wpam"
 	"socialpredict/models"
 	"socialpredict/setup"
+	"socialpredict/util"
+	"sort"
 	"testing"
 	"time"
 )
 
 var now = time.Now()
 
-func TestAdjustPayoutsFromNewest(t *testing.T) {
+// helper function to create wpam.ProbabilityChange points succiently
+func generateProbability(probabilities ...float64) []wpam.ProbabilityChange {
+	var changes []wpam.ProbabilityChange
+	for _, p := range probabilities {
+		changes = append(changes, wpam.ProbabilityChange{Probability: p})
+	}
+	return changes
+}
+
```

Comment:
I think this is a good candidate to include in our modelstesting package. We would only ever generate a bet this way when we're testing, so it would be good to make it available there.